docs/docs/introducing-jsx.md

```diff
@@ -0,0 +1,156 @@
+---
+id: introducing-jsx
+title: Introducing JSX
+permalink: docs/introducing-jsx.html
+prev: hello-world.html
+next: rendering-elements.html
+---
+
+Consider this variable declaration:
+
+```js
+const element = <h1>Hello, world!</h1>;
+```
+
+This funny tag syntax is neither a string nor HTML.
+
+It is called JSX, and it is a syntax extension to JavaScript. We recommend using it with React to describe what the UI should look like. JSX may remind you of a template language, but it comes with the full power of JavaScript.
+
+### Embedding Expressions in JSX
+
+JSX comes with the full power of JavaScript.
+
+You can embed any [JavaScript expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators#Expressions) in JSX by wrapping it in curly braces.
+
+For example, `2 + 2`, `user.name`, and `formatName(user)` are all valid expressions:
+
+```js{12}
+function formatName(user) {
+  return user.firstName + ' ' + user.lastName;
+}
+
+const user = {
+  firstName: 'Harper',
+  lastName: 'Perez'
+};
+
+const element = (
+  <h1>
+    Hello, {formatName(user)}!
+  </h1>
+);
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/PGEjdG?editors=0010)
+
+We wrapped JSX in parentheses and split it over multiple lines for readability. This also helps avoid the pitfalls of [automatic semicolon insertion](http://stackoverflow.com/q/2846283).
+
+### JSX is an Expression Too
+
+After compilation, JSX expressions become regular JavaScript objects.
+
+This means that you can use JSX inside of `if` statements and `for` loops, assign it to variables, accept it as arguments, and return it from functions:
+
+```js{3,5}
+function getGreeting(user) {
+  if (user) {
+    return <h1>Hello, {formatName(user.name)}!</h1>;
+  } else {
+    return <h1>Hello, Stranger.</h1>;
+  }
+}
+```
+
+### Specifying Attributes with JSX
+
+You may use quotes to specify string literals as attributes:
+
+```js
+const element = <div tabIndex="0"></div>;
+```
+
+You may also use curly braces to embed a JavaScript expression in an attribute:
+
+```js
+const element = <img src={user.avatarUrl}></img>;
+```
+
+### Specifying Children with JSX
+
+If a tag is empty, you may close it immediately with `/>`, like XML:
+
+```js
+const element = <img src={user.avatarUrl} />;
+```
+
+JSX tags may contain children:
+
+```js
+const element = (
+  <div>
+    <h1>Hello!</h1>
+    <h2>Good to see you here.</h2>
+  </div>
+);
+```
+
+>**Caveat:**
```

Comment:
Yea, I have a todo item for looking through all docs and cross-linking them when everything is in place